server/sonar-server/src/main/java/org/sonar/server/qualityprofile/index/ActiveRuleIndex.java

```diff
@@ -19,179 +19,211 @@
+import static org.sonar.server.rule.index.RuleIndexDefinition.FIELD_RULE_STATUS;
+import static org.sonar.server.rule.index.RuleIndexDefinition.INDEX;
+import static org.sonar.server.rule.index.RuleIndexDefinition.TYPE_ACTIVE_RULE;
+import static org.sonar.server.rule.index.RuleIndexDefinition.TYPE_RULE;
+
+/**
+ * The unique entry-point to interact with Elasticsearch index "active rules".
+ * All the requests are listed here.
+ */
+public class ActiveRuleIndex extends BaseIndex {
 
   public static final String COUNT_ACTIVE_RULES = "countActiveRules";
 
-  public ActiveRuleIndex(ActiveRuleNormalizer normalizer, SearchClient node) {
-    super(IndexDefinition.ACTIVE_RULE, normalizer, node);
+  public ActiveRuleIndex(EsClient client) {
+    super(client);
   }
 
-  @Override
-  protected String getKeyValue(ActiveRuleKey key) {
-    return key.toString();
-  }
-
-  @Override
-  protected Map mapKey() {
-    Map<String, Object> mapping = new HashMap<>();
-    mapping.put("path", ActiveRuleNormalizer.ActiveRuleField.KEY.field());
-    return mapping;
-  }
-
-  @Override
-  protected Map mapProperties() {
-    Map<String, Object> mapping = new HashMap<>();
-    for (IndexField field : ActiveRuleNormalizer.ActiveRuleField.ALL_FIELDS) {
-      mapping.put(field.field(), mapField(field));
+  /**
+   * @deprecated since 5.5, use {@link org.sonar.db.qualityprofile.ActiveRuleDao instead}
+   */
+  @Deprecated
+  @CheckForNull
+  public ActiveRule getNullableByKey(ActiveRuleKey key) {
+    GetRequestBuilder request = getClient().prepareGet()
+      .setIndex(RuleIndexDefinition.INDEX)
+      .setType(RuleIndexDefinition.TYPE_ACTIVE_RULE)
+      .setId(key.toString())
+      .setFetchSource(true)
+      .setRouting(key.ruleKey().repository());
+
+    GetResponse response = request.get();
+    if (response.isExists()) {
+      return new ActiveRuleDoc(response.getSource());
     }
-    return mapping;
-  }
-
-  @Override
-  protected Map mapDomain() {
-    Map<String, Object> mapping = new HashMap<>();
-    mapping.put("dynamic", false);
-    mapping.put("_id", mapKey());
-    mapping.put("_parent", mapParent());
-    mapping.put("_routing", mapRouting());
-    mapping.put("properties", mapProperties());
-    return mapping;
-  }
-
-  private Map mapRouting() {
-    Map<String, Object> mapping = new HashMap<>();
-    mapping.put("required", true);
-    mapping.put("path", ActiveRuleNormalizer.ActiveRuleField.RULE_KEY.field());
-    return mapping;
-  }
-
-  private Object mapParent() {
-    Map<String, Object> mapping = new HashMap<>();
-    mapping.put("type", this.getParentType());
-    return mapping;
-  }
-
-  @Override
-  public ActiveRule toDoc(Map<String, Object> fields) {
-    return new ActiveRuleDoc(fields);
+    return null;
   }
 
   /**
-   * finder methods
+   * @deprecated since 5.5, use {@link org.sonar.db.qualityprofile.ActiveRuleDao instead}
    */
+  @Deprecated
   public List<ActiveRule> findByRule(RuleKey key) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)